src/dataflow/src/source/s3.rs

```diff
@@ -820,19 +814,13 @@ where
                     .await
                     .is_err()
                 {
-                    return (DownloadStatus::SendFailed, None);
+                    return Err(DownloadError::SendFailed);
                 }
             }
             Err(_) => {
-                return (
-                    DownloadStatus::Failed {
-                        err: S3Error::RetryFailed,
-                    },
-                    Some(DownloadMetricUpdate {
-                        bytes: bytes_read.try_into().expect("usize <= u64"),
-                        messages: chunks,
-                    }),
-                )
+                return Err(DownloadError::Failed {
+                    err: S3Error::RetryFailed,
```

Comment:
I think in this case `S3Error::RetryFailed` isn't getting us anything, and it'd be better to change the `err` field  `cause: Box<dyn std::error::Error>` and put the `Err` that we're currently throwing away into it.

Reply:
I think its actually an `std::io::Error`? Ill clean it up

Reply:
there is a bit of an awkward back and forth with the channel, but I put up what I came up with

Reply:
> I think its actually an std::io::Error?

Yeah that makes sense.

> there is a bit of an awkward back and forth with the channel, but I put up what I came up with

Oh yeah I see.